665 & 666 - Map Layers Updates

JessicaInTime opened the pull request:
bcgov/BCHeritage#665 & bcgov/BCHeritage#666

This includes the changes for the mapbox layers on BCRHP and the map popup content.

JrodS committed:
665 - Fix migration dependency, fix sql query, added labels to islands trust and tourism overlays

JrodS committed:
665 - Added new mapbox layers for bcrhp and updating public visibility for existing layers to the export json files

From this point, new and updating mapbox layers will be from the .json files and then deploying with a django migration with running the arches cmd for adding and deleting mapbox layers

JrodS committed:
665 - Added migration modules for changes to bcrhp mapbox layers

JrodS committed:
666 - Added/changed objects for map popup display for bcrhp

JrodS committed:
665 - Fixed errors for bcrhp mapbox layers and bad name for migration module

JrodS committed:
665 - Fixed boundary and fill colour on bcrhp mapbox layers

JrodS committed:
Merge 665/666 changes into clean branch

JrodS committed:
665 - Added zoom levels on new mapbox layers for bcrhp

JrodS committed:
665 - Added json mapbox layers to bcrhp from common

JrodS committed:
665 - Added migration for public visibility setting for requested mapbox layers on bcrhp

JrodS committed:
665 - Replace ansible { url_context_root } var with hardcoded deployed internal url /int/bcrhp/...

JrodS committed:
665 - Changed the remove requested mapbox layers in bcrhp migration

JrodS committed:
665 - Added sort order to all bcrhp mapbox layers

665 - Added sort order to all bcrhp mapbox layers

JrodS committed:
665 - Updated migration module names to have the ticket number as the prefix, same to what arches core does

Apparently the privileged_sql.py module wasn't in bcrhp and only common

bferguso commented on src/bcrhp/bcrhp/migrations/665_4_public_internal_map_layers.py:
```diff
@@ -0,0 +1,46 @@
+
+def backward(apps, schema_editor):
+    db_alias = schema_editor.connection.alias
+    MapLayer = apps.get_model('arches', 'MapLayer')
```
Is there a reason you're not using the privileged SQL method for the backward migration? This also feels like the reverse should be the same as the forward but from the common package directory.

JessicaInTime replied:
The reason for not using the privileged sql is because of the way the migration operation objects work, since im running python code, the privileged sql class isn't suitable.

The reason for the django model usage is due to the nature of the json files being static. For reverting the data of the map layer, the json files would need to be the previous data but that doesn't work when you need to be on this commit for example to have the new changes to go forward.

Since the get_model call isn't working, what i might be able to do is call a raw execute sql function with the username and pass as what the privileged sql is doing.

bferguso replied:
OK, great. Thank you for clarifying.

bferguso commented on src/bcrhp/bcrhp/pkg/map_layers/mapbox_spec_json/overlays/nts250k/nts250k.json:
```diff
@@ -0,0 +1,52 @@
+{
```
This layer is to be removed so we probably don't need to copy it over.

JessicaInTime replied:
The reason why this is copied over is for the backward migration. Same with the other layers that will be removed

bferguso commented on src/bcrhp/bcrhp/pkg/map_layers/mapbox_spec_json/overlays/nts50k/nts50k.json:
```diff
@@ -0,0 +1,54 @@
+{
```
This layer is to be removed so we probably don't need to copy it over.

bferguso replied:
Agreed that we'll leave it here for now until next release. Need a PR to remove the unused map layers in the next release.

bferguso commented on src/bcrhp/bcrhp/pkg/map_layers/mapbox_spec_json/overlays/provinces-states/provinces-states.json:
```diff
@@ -0,0 +1,37 @@
+{
```
This layer is to be removed so we probably don't need to copy it over.

bferguso commented on src/bcrhp/bcrhp/pkg/map_layers/mapbox_spec_json/overlays/trim-contours/trim-contours.json:
```diff
@@ -0,0 +1,44 @@
+{
```
This layer is to be removed so we probably don't need to copy it over.

bferguso commented on src/bcrhp/bcrhp/pkg/map_layers/mapbox_spec_json/overlays/rivers/rivers.json:
```diff
@@ -0,0 +1,63 @@
+    "isoverlay": true,
+    "legend": "<p><img alt=\"\" class=\"rivers legend-swatch\" src=\"/int/bcrhp/static/img/blank.png\" />Rivers</p>",
+    "searchonly": false,
+    "sortorder": 0,
```
This is showing up first. Should check with emerald if they still want it and if so where it should be in the list.

JessicaInTime replied:
I will ask on the ticket.

bferguso commented on src/bcrhp/bcrhp/pkg/map_layers/mapbox_spec_json/overlays/streams/streams.json:
```diff
@@ -0,0 +1,52 @@
+    "isoverlay": true,
+    "legend": "<p><img alt=\"\" class=\"streams legend-line\" src=\"/int/bcrhp/static/img/blank.png\" />Streams</p>",
+    "searchonly": false,
+    "sortorder": 0,
```
This is showing up first. Should check with emerald if they still want it and if so where it should be in the list.

bferguso replied:
Thanks for fixing!

bferguso commented on src/bcrhp/bcrhp/pkg/map_layers/mapbox_spec_json/overlays/indian-reserves-band-names/indian-reserves-band-names.json:
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "Indian Reserves and Band Names",
```
Please update with name requested in the ticket.

JessicaInTime replied:
I will update, I missed the edit that was made on the ticket for this specific detail.

bferguso reviewed (changes requested):
I think a bunch of the overlay files shouldn't be copied over as they're slated for removal. Also - please confirm what should be done with streams and river layers (keep/remove and if keeping, what order they should be in).

bferguso commented on src/bcrhp/bcrhp/migrations/operations/privileged_sql.py:
```diff
@@ -0,0 +1,42 @@
+from django.db.migrations.operations.special import RunSQL
```
Was this changed from the version in `/common/migrations/operations`? If not, please remove.

JessicaInTime replied:
My bad, having some logic only in common throws me off.

JessicaInTime committed:
Delete src/bcrhp/bcrhp/migrations/operations/privileged_sql.py

JrodS committed:
665 - Addressed PR feedback on migration for updating bcrhp public visibility map layers

JrodS committed:
665 - Fixed mapbox layer name for bcrhp indian reserves to indigenous reserves

JrodS committed:
665 - Fixed bad file path naming on the natural resource regions mapbox layer

JrodS committed:
665 - Fixed a few issues on bcrhp for map layer migrations:

- Added city to map popup nodes
- Removed comment for legal description, is present on databc
- Changed starting migration module dependency away from other active task
- Fixed url prefix for indian reserves json tiles

JrodS committed:
665 - Adjust public visibility for additional bcrhp map layers

JrodS committed:
665 - Updated sort order rivers and streams bcrhp map layers

bferguso reviewed (approved):
Looking good! We'll need to verify the content on DLVR.

bferguso committed:
665 - Fix island trust label layer
